File: Frontend/Frontend/src/app/services/category.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';
import { CategoryCreateRequest, CategoryResponse } from '../dtos/category.dto';
import { Page } from '../models/page';

@Injectable({
  providedIn: 'root'
})
export class CategoryService {
  private readonly apiUrl = `${environment.apiUrl}/categories`;

  constructor(private http: HttpClient) {}

  list(page: number = 0, size: number = 20): Observable<Page<CategoryResponse>> {
    const params = new HttpParams()
      .set('page', page.toString())
      .set('size', size.toString());

    return this.http.get<Page<CategoryResponse>>(this.apiUrl, { params });
  }

  create(request: CategoryCreateRequest): Observable<CategoryResponse> {
    return this.http.post<CategoryResponse>(this.apiUrl, request);
  }

  update(id: number, request: any): Observable<CategoryResponse> {
    return this.http.put<CategoryResponse>(`${this.apiUrl}/${id}`, request);
  }

  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}

File: elearning-platform/elearning-platform/src/main/java/com/elearning/elearning_platform/category/service/CategoryService.java
package com.elearning.elearning_platform.category.service;

import com.elearning.elearning_platform.category.dto.*;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface CategoryService {
    CategoryResponse create(String actorRole, CategoryCreateRequest req);
    CategoryResponse update(String actorRole, Long id, CategoryUpdateRequest req);
    void delete(String actorRole, Long id);
    Page<CategoryResponse> list(Pageable pageable);
}